conformance/lang/expressions/mapping_constructor_expr.balt

```diff
@@ -0,0 +1,1917 @@
+        record-type, string
+
+function init() {
+    int c = 45;
+    byte e = 128;
+
+    printResult({a: 325, b: "value b", c, d: (), e}); // @output {"a":325,"b":"value b","c":45,"d":null,"e":128}
+    printResult({a: 325, b: "value b", c, d: (), e} == {a: 325, b: "value b", c, d: (), e: e}); // @output true
+ 
+    record {|int c; byte e;|} result = {c, e};
+    printResult(result); // @output {"c":45,"e":128}
+    printResult({c, e} == {c: c, e: e}); // @output true
+
+    float f = 55.43;
+    map<any> result2 = {a: "Some string", b: 434, result, f};
+    printResult(result2); // @output {"a":"Some string","b":434,"result":{"c":45,"e":128},"f":55.43}
+    printResult({a: "Some string", b: 434, result, f} == {a: "Some string", b: 434, result: result, f: f}); // @output true
+}
+
+function printResult(any result) {
+    io:println(result);
+}
+
+Test-Case: output
+Description: Test basic mapping constructor expression with rest fields.
+Labels: any, byte, mapping-constructor-expr, map-type, string
+
+function init() {
+    int a = 45;
+    byte b = 128;
+
+    printResult({fieldA: true, "a": "fdfd", "b": 43}); // @output {"fieldA":true,"a":"fdfd","b":43}
+
+    printResult({"a": "fdfd", "b": 43}); // @output {"a":"fdfd","b":43}
+
+    map<any> mapVar = {"a": "fdfd", fieldB: "Some string", "b": 43, fieldC: false};
+    printResult(mapVar); // @output {"a":"fdfd","fieldB":"Some string","b":43,"fieldC":false}
+
+    printResult({a, fieldB: "Some string", "b": b, fieldC: false}); // @output {"a":45,"fieldB":"Some string","b":128,"fieldC":false}
+}
+
+function printResult(map<any> result) {
+    io:println(result);
+}
+
+Test-Case: output
+Description: Test basic mapping constructor expression with readonly specific fields.
+Labels: any, mapping-constructor-expr, map-type, readonly-field, string
+
+function init() {
+    int a = 45;
+
+    printResult({readonly fieldA: true, "a": "fdfd", readonly "b": 43}); // @output {"fieldA":true,"b":43,"a":"fdfd"}
+
+    printResult({"a": "fdfd", readonly "b": 43}); // @output {"b":43,"a":"fdfd"}
+
+    printResult({A: 325, b: "value b", readonly a}); // @output {"a":45,"A":325,"b":"value b"}
+    io:println({A: 325, b: "value b", readonly a} == {A: 325, b: "value b", readonly a: a}); // @output true
+}
+
+function printResult(map<any> result) {
+    io:println(result);
+}
+
+Test-Case: error
+Description: Test Duplicate specific fields are not allowed.
+Labels: any, mapping-constructor-expr, map-type, string
+
+function errorFunction() {
+    map<any> _ = {fieldB1: 454, fieldB1: "some string"}; // @error Duplicate key `fieldB1`
+    map<any> _ = {_fieldB1: 454, _fieldB1: "some string"}; // @error Duplicate key `_fieldB1`
+    
+    map<any> _ = {fieldB1: 454, field\u{0042}1: 454}; // @error Duplicate key `fieldB1`
+    map<any> _ = {field\u{0042}1: 454, field\u{0042}1: 454}; // @error Duplicate key `fieldB1`
```

Comment:
Here again let's split up tests to be as concise as possible.

Reply:
These tests are only using specific fields and map types so it's okay to keep it as one test right?